### ISSUE TITLE
Using psalm in phtml files

### ISSUE THREAD

**Issue by quisse** [vimeo/psalm#4162]
I'm trying to make psalm work on `.phtml` files.
Everything goes well, except for one thing. All classes (which should be autoloaded) are undefined. In normal php classes they get loaded by psalm.

Following throws `UndefinedDocblockClass` for `CustomBlocks`.

    //my-template.phtml
    
    <?php 
    use MyNamespace\Block\CustomBlocks

    /** @var CustomBlocks $block */
    ?>

**Comment by quisse** [comment]
@psalm-github-bot No, not possible.

**Comment by weirdan** [comment]
Can you provide a minimal reproducer as a separate public repository?

**Comment by quisse** [comment]
https://github.com/storefront-bvba/psalm-phtml-issue

**Comment by weirdan** [comment]
Confirmed on 3.15 and dev-master, with these fixes applied:
```patch
diff --git a/composer.json b/composer.json
index b56fcb1..16e2ac3 100644
--- a/composer.json
+++ b/composer.json
@@ -9,9 +9,7 @@
             "Magento\\Framework\\": "lib\/internal\/Magento\/Framework\/",
             "Magento\\Setup\\": "setup\/src\/Magento\/Setup\/",
             "Magento\\": "app\/code\/Magento\/",
-            "Zend\\Mvc\\Controller\\": "setup\/src\/Zend\/Mvc\/Controller\/"
-        },
-        "psr-4": {
+            "Zend\\Mvc\\Controller\\": "setup\/src\/Zend\/Mvc\/Controller\/",
             "MyNamespace\\Blocks\\":"src"
         }
     }
diff --git a/psalm.xml b/psalm.xml
index 3240886..3b1980a 100644
--- a/psalm.xml
+++ b/psalm.xml
@@ -8,6 +8,8 @@
 >
     <projectFiles>
         <directory name="src" />
+	<file name="good.phtml" />
+	<file name="bad.phtml" />
         <ignoreFiles>
             <directory name="vendor" />
         </ignoreFiles>
diff --git a/src/CustomBlocks.php b/src/CustomBlocks.php
index e69de29..df1d752 100644
--- a/src/CustomBlocks.php
+++ b/src/CustomBlocks.php
@@ -0,0 +1,5 @@
+<?php
+
+namespace MyNamespace\Blocks;
+
+class CustomBlocks {}
```

**Comment by quisse** [comment]
Woops, went too fast through commit. Applied patch. Thanks for checking!

**Comment by weirdan** [comment]
The reason why `good.phtml` does not fail is because the docblock is associated with the following class, and `@var` is ignored in class docblocks. Adding a semicolon after the `/** @var CustomBlocks $block */`:
```php
`/** @var CustomBlocks $block */
;
class Hey {}
```
surfaces the same issue.

**Comment by quisse** [comment]
Oh ok, clear. Do I need to adjust the reproducer?

**Comment by weirdan** [comment]
You can drop `good.phtml`

**Comment by quisse** [comment]
Done

**Comment by quisse** [comment]
Adding root as an argument solves the issue.

**Comment by quisse** [comment]
Instantiating an object with the class which throws the `UndefinedDocblockClass` makes that the class get loaded & solves the error. So afaik the issue must be somewhere in https://github.com/vimeo/psalm/blob/43f6921efa4c26f00738b4cfc5188239aa05797a/src/Psalm/Internal/PhpVisitor/ReflectorVisitor.php#L125-L407 but that's one helluva method 🧐

**Comment by quisse** [comment]
solved in f3cc7a7f2fb9611abe5db5462d7d0ff8d3d9d164
